src/libraries/System.Private.CoreLib/src/System/IO/BufferedStream.cs

```diff
@@ -1164,6 +1305,18 @@ public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationTo
             TaskToApm.End(asyncResult);
 
         public override void WriteByte(byte value)
+        {
+            if (_writePos > 0 && _writePos < _bufferSize - 1)
```

Comment:
Does closing the stream guarantee _writePos is set to 0?

Reply:
It does not. I am going to ensure it does.